commit 8d560802760494390d53f7adc1b36ab04eaf6c68
Author: Jim Desrivieres <jdesrivieres>
Date:   Sat Nov 15 02:55:57 2003 +0000

    Improve spec for statements (bug 39739)

3	1	org.eclipse.jdt.core/buildnotes_jdt-core.html
8	0	org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/DoStatement.java
8	0	org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/ForStatement.java
16	0	org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/IfStatement.java
8	0	org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/LabeledStatement.java
8	0	org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/WhileStatement.java
diff --git a/org.eclipse.jdt.core/buildnotes_jdt-core.html b/org.eclipse.jdt.core/buildnotes_jdt-core.html
index a02c424..18b8c3c 100644
--- a/org.eclipse.jdt.core/buildnotes_jdt-core.html
+++ b/org.eclipse.jdt.core/buildnotes_jdt-core.html
@@ -48,6 +48,8 @@ What's new in this drop</h2>
 <h3>Problem Reports Fixed</h3>
 <a href="http://bugs.eclipse.org/bugs/show_bug.cgi?id=14103">14103</a>
 [Builder] Too many dependents found when incrementally recompiling
+<br><a href="http://bugs.eclipse.org/bugs/show_bug.cgi?id=39739">39739</a>
+[DOM/AST] VariableDeclarationStatements aren't full statements
 <br><a href="http://bugs.eclipse.org/bugs/show_bug.cgi?id=46690">46690</a>
 Code formatter always inserts space after comma in multiple locals or field declarations
 <br><a href="http://bugs.eclipse.org/bugs/show_bug.cgi?id=46689">46689</a>
@@ -57,7 +59,7 @@ Code formatter doesn't indent properly statements following a switch statement
 <br><a href="http://bugs.eclipse.org/bugs/show_bug.cgi?id=46356">46356</a>
 [Builder] simple/qualified names list for indicting dependents should be hashed collections
 <br><a href="http://bugs.eclipse.org/bugs/show_bug.cgi?id=46612">46612</a>
-BodyDeclaration should provide a method getModifiers
+[DOM/AST] BodyDeclaration should provide a method getModifiers
 
 <a name="v_382"></a>
 <p><hr><h1>
diff --git a/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/DoStatement.java b/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/DoStatement.java
index 6903a0e..f898130 100644
--- a/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/DoStatement.java
+++ b/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/DoStatement.java
@@ -140,6 +140,14 @@ public class DoStatement extends Statement {
 	
 	/**
 	 * Sets the body of this do statement.
+	 * <p>
+	 * Special note: The Java language does not allow a local variable declaration
+	 * to appear as the body of a do statement (they may only appear within a
+	 * block). However, the AST will allow a <code>VariableDeclarationStatement</code>
+	 * as the body of a <code>DoStatement</code>. To get something that will
+	 * compile, be sure to embed the <code>VariableDeclarationStatement</code>
+	 * inside a <code>Block</code>.
+	 * </p>
 	 * 
 	 * @param statement the body statement node
 	 * @exception IllegalArgumentException if:
diff --git a/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/ForStatement.java b/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/ForStatement.java
index 6ae1220..cf6eaf9 100644
--- a/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/ForStatement.java
+++ b/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/ForStatement.java
@@ -193,6 +193,14 @@ public class ForStatement extends Statement {
 	
 	/**
 	 * Sets the body of this for statement.
+	 * <p>
+	 * Special note: The Java language does not allow a local variable declaration
+	 * to appear as the body of a for statement (they may only appear within a
+	 * block). However, the AST will allow a <code>VariableDeclarationStatement</code>
+	 * as the body of a <code>ForStatement</code>. To get something that will
+	 * compile, be sure to embed the <code>VariableDeclarationStatement</code>
+	 * inside a <code>Block</code>.
+	 * </p>
 	 * 
 	 * @param statement the body statement node
 	 * @exception IllegalArgumentException if:
diff --git a/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/IfStatement.java b/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/IfStatement.java
index fd6c685..7aef182 100644
--- a/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/IfStatement.java
+++ b/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/IfStatement.java
@@ -150,6 +150,14 @@ public class IfStatement extends Statement {
 	
 	/**
 	 * Sets the "then" part of this if statement.
+	 * <p>
+	 * Special note: The Java language does not allow a local variable declaration
+	 * to appear as the "then" part of an if statement (they may only appear within a
+	 * block). However, the AST will allow a <code>VariableDeclarationStatement</code>
+	 * as the thenStatement of a <code>IfStatement</code>. To get something that will
+	 * compile, be sure to embed the <code>VariableDeclarationStatement</code>
+	 * inside a <code>Block</code>.
+	 * </p>
 	 * 
 	 * @param statement the "then" statement node
 	 * @exception IllegalArgumentException if:
@@ -189,6 +197,14 @@ public class IfStatement extends Statement {
 	 * (as in <code>"if(true){}"</code>) and having an empty block (as in
 	 * "if(true){}else{}") or null statement (as in "if(true){}else;"). 
 	 * </p>
+	 * <p>
+	 * Special note: The Java language does not allow a local variable declaration
+	 * to appear as the "else" part of an if statement (they may only appear within a
+	 * block). However, the AST will allow a <code>VariableDeclarationStatement</code>
+	 * as the elseStatement of a <code>IfStatement</code>. To get something that will
+	 * compile, be sure to embed the <code>VariableDeclarationStatement</code>
+	 * inside a <code>Block</code>.
+	 * </p>
 	 * 
 	 * @param statement the "else" statement node, or <code>null</code> if 
 	 *    there is none
diff --git a/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/LabeledStatement.java b/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/LabeledStatement.java
index 5083b6d..80aebe4 100644
--- a/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/LabeledStatement.java
+++ b/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/LabeledStatement.java
@@ -140,6 +140,14 @@ public class LabeledStatement extends Statement {
 	
 	/**
 	 * Sets the body of this labeled statement.
+	 * <p>
+	 * Special note: The Java language does not allow a local variable declaration
+	 * to appear as the body of a labeled statement (they may only appear within a
+	 * block). However, the AST will allow a <code>VariableDeclarationStatement</code>
+	 * as the body of a <code>LabeledStatement</code>. To get something that will
+	 * compile, be sure to embed the <code>VariableDeclarationStatement</code>
+	 * inside a <code>Block</code>.
+	 * </p>
 	 * 
 	 * @param statement the body statement node
 	 * @exception IllegalArgumentException if:
diff --git a/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/WhileStatement.java b/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/WhileStatement.java
index c6ecb02..38338bf 100644
--- a/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/WhileStatement.java
+++ b/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/WhileStatement.java
@@ -141,6 +141,14 @@ public class WhileStatement extends Statement {
 	
 	/**
 	 * Sets the body of this while statement.
+	 * <p>
+	 * Special note: The Java language does not allow a local variable declaration
+	 * to appear as the body of a while statement (they may only appear within a
+	 * block). However, the AST will allow a <code>VariableDeclarationStatement</code>
+	 * as the body of a <code>WhileStatement</code>. To get something that will
+	 * compile, be sure to embed the <code>VariableDeclarationStatement</code>
+	 * inside a <code>Block</code>.
+	 * </p>
 	 * 
 	 * @param statement the body statement node
 	 * @exception IllegalArgumentException if:
